### ISSUE TITLE
Add possibility to configure multiple outputs for screen recording

### ISSUE THREAD

**Issue by pdluke** [ShareX/ShareX#1721]
In the screen recording options it would be nice to configure multiple outputs, like creating a gif and a mp4 file at once.
I know that it should be possible to configure it with a custom ffmpeg command, but for normal users it would be way more convenient.

**Comment by Jaex** [comment]
I'm not planning to add multiple output for screen recording.

**Comment by pdluke** [comment]
You mind explaining why? I posted this here for people to discuss, because GitHub does not have a feature like Uservoice to vote for demand.
I find it somehow rude to instantly close the issue.

**Comment by dannoe** [comment]
I agree with Jaex. 
Its a very specific use case. And as you mentioned you can probably achieve it with custom commands already. 
Another solution would be, to record it as mp4 and then use custom uploaders to upload it to some mp4-to-gif Converter online.

**Comment by Jaex** [comment]
I also find it rude every single time I tell that I don't want to implement something there is always people find me rude and tells stuffs to me. It is like I am forced to implement every single feature people want.

**Comment by pdluke** [comment]
Did I demand that especially YOU have to implement it? This is an open-source project, anybody could implement it, not only YOU. You are the project maintainer, at the end you handle the pull request.
Why not users decide what feature is useful and what not. But blocking the discussion from the beginning, because YOU do not want to implement is no proper way to go with the community. What would have cost it to you to just leave it open so people would find it easier, instead of closing it immediately. I find your way of discussion rude not that you don't want to implement it. Are you that arrogant to decide everything for the community what is good are not? Sorry for the hard words, but this really grinds my gears.

**Comment by Jaex** [comment]
Your request was very specific to your needs and I don't think it will be useful for anyone else this is why I didn't want to implement it or anyone to implement it. Not to mention doing multiple output recording can slow down encoding (recording) greatly.

What gonna change if I wrote these reasons before close? Other than wasting my time because now I wasted my 5 minutes to write this message because English far more difficult than C#. There is nothing to do with being arrogant I already left so many features open in GitHub issues that I don't want to implement myself, because I found that features could be benefit for more people but not this one.

I also didn't block any discussion people can still discuss in closed issue.

**Comment by pdluke** [comment]
Yeah it is a very special case, this is why I wanted to discuss it. It is your valid opinion to think that it will not be useful, no problem with that you express it. But think how people feel that make a suggestion just to get the door instantly slapped before their faces. I expected to have an productive and open discussion with people at least the developers. Maybe not today but over time. Yeah the discussion is not blocked entirely, but default github filter is for open issues, so most people will not notice it.
The information you mentioned about possible performance issues is that kind of discussion I was looking for as I did not consider it.
English is also not my first language as it is for you but it's no excuse to avoid dealing with discussion by closing the issue prematurely.

But never mind, if you think this feature is bullshit and people don't need than I leave it be with his message. You are the developer and know way better what people want. I'm just someone trying to make helpful suggestions.

**Comment by mattrice** [comment]
Apologies for the thread necromancy: it is possible to configure ShareX to output both an MP4 and a GIF for a single screen recording [using ShareX Actions](https://getsharex.com/actions/#convert-video-to-gif). I have a hotkey that starts/stops screen record (to MP4) with an after capture action that converts the MP4 to GIF; I configured an action as per the documentation page.

The arguments example was particularly helpful, as I expect it would have taken me a very long time to come up with something similar:
```
-i "$input" -lavfi "palettegen=stats_mode=full[palette],[0:v][palette]paletteuse=dither=sierra2_4a" -y "$output"
```

It is slightly hard-drive wasteful to output both files, and the GIF conversion is not as immediate, but automatically creating both gives me choices.

I created this GIF using this exact method!
![2022-03-10T134201Z516](https://user-images.githubusercontent.com/3780536/157732895-b65c5273-0dd4-403f-9c59-0c297d6a3c9c.gif)
